source/javascripts/services/_appService.js.erb

```diff
@@ -217,28 +225,51 @@ angular.module("BitriseWorkflowEditor").service("appService", function($q, reque
 			normalizeSecretConfigs(secretConfigs);
 
 			appService.secrets = _.map(secretConfigs, function(aSecretConfig) {
-				return new Variable(aSecretConfig, Variable.defaultVariableConfig());
+				return new Variable(aSecretConfig, Variable.defaultVariableConfig(), false, false);
 			});
 
 			appService.savedSecrets = angular.copy(appService.secrets);
 		});
 	};
 
+	appService.getSecretEnvValue = function(envVar, requestConfig) {
+		var secretEnvVar = envVar;
+		if (!secretEnvVar && appService.secrets) {
+			return $q.when();
+		}
+
+		return requestService.getSecretEnvValue(secretEnvVar, requestConfig).then(function(secretValue) {
+			var envVar = _.find(appService.secrets, function(aSecret) {
```

Comment:
No need for a re-find. You already have the same instance - from the original function's arguments.